app/views/users/show.html.erb

```diff
@@ -0,0 +1,13 @@
+<h1>Welcome <%= @user.name %> </h1>
+
+<h3>Previously attended events</h3>
+<% @user.attended_events.where("date < ?", Date.today).each do |event|%>
+  <%= link_to event.description, event_path(event.id) %><br>
+<% end %>
+
+<h3>Upcoming events</h3>
+<% @user.attended_events.where("date >= ?", Date.today).each do |event|%>
+  <%= link_to event.description, event_path(event.id) %><br>
+<% end %>
```

Comment:
The better to do it it's by using Event scope
```rb
@user.attended_events.upcoming
```
Create a variable in your action controller and use it here (in that view)
```rb
def show
  @upcoming = @user.attended_events.upcoming
  ...
  ...
end
```